TCP_Server.py

```diff
@@ -97,8 +97,11 @@ def close_connection(self, conn):
             return
         peername = self.current_peers[conn.get().fileno()]
         logging.info('closing connection to {0}'.format(peername))
-        del self.current_peers[conn.get().fileno()]
-        self.selector.unregister(conn.get())
+        try:
+            del self.current_peers[conn.get().fileno()]
+            self.selector.unregister(conn.get())
```

Comment:
You can always check first if peer is on list and then delete, but this is fine.
Just add which peer is missing to log.
"no peer" message says nothing :)